lib/ufolint/validators/plistvalidators.py

```diff
@@ -289,29 +289,29 @@ def run_ufolib_import_validation(self):
                 # glyphs_dir_list is a list of lists mapped to glyphs dir name,
                 # glyphs dir path
                 gs = GlyphSet(
-                    str(rel_dir_path), ufoFormatVersion=self.ufoversion, validateRead=True
+                    str(rel_dir_path),
+                    ufoFormatVersion=self.ufoversion,
+                    validateRead=True,
+                    expectContentsFile=True,
                 )  # test for raised exceptions
                 res.test_failed = False
 
-                # Check for unlisted files if contents.plist exists. It is mandated by
-                # the spec, but if it does not exist, glifLib will shrug and say the
-                # glyph set was empty.
-                if (rel_dir_path / self.testfile).exists():
-                    files = set(gs.contents.values())
-                    files.update(("contents.plist", "layerinfo.plist"))
-                    files_actually = set(
-                        str(p.relative_to(rel_dir_path))
-                        for p in rel_dir_path.glob("**/*")
+                # Check for unlisted files.
+                files = set(gs.contents.values())
+                files.update(("contents.plist", "layerinfo.plist"))
+                files_actually = set(
+                    str(p.relative_to(rel_dir_path))
+                    for p in rel_dir_path.glob("**/*")
+                )
+                unlisted_files = files_actually - files
+                if unlisted_files:
+                    res.test_failed = True
+                    res.exit_failure = True  # sacrilege
```

Comment:
How do you suggest that we refactor the sacrilege here?

Reply:
Oh, no, I consider unregistered glif files a sacrilege that merits an error even when it doesn't prevent compilation.